docs/public/components/testing.md

```diff
@@ -1,3 +1,203 @@
+
+=== ":octicons-file-code-16: comp/someomponent/impl/component.go"
+    ```go
+    package impl
+
+    import (
+      "context"
+      
+      someomponent "github.com/DataDog/datadog-agent/comp/someomponent/def"
+      compdef "github.com/DataDog/datadog-agent/comp/def"
+    )
+
+    type Requires struct {
+      Lc      compdef.Lifecycle
+    }
+
+    type Provides struct {
+      Comp someomponent.Component
+    }
+
+    type component struct {
+      started  bool
+      stopped bool
+    }
+
+    func (c *component) start() error {
+      // [...]
+      
+      c.started = true
+
+      return nil
+    }
+
+    func (h *healthprobe) stop() error {
+      // [...]
+      
+      c.stopped = true
+      c.started = false
+
+      return nil
+    }
+
+    // NewComponent creates a new healthprobe component
+    func NewComponent(reqs Requires) (Provides, error) {
+      provides := Provides{}
+      comp := &component{}
+
+      reqs.Lc.Append(compdef.Hook{
+        OnStart: func(ctx context.Context) error {
+          return comp.start()
+        },
+        OnStop: func(ctx context.Context) error {
+          return comp.stop()
+        },
+      })
+
+      provides.Comp = comp
+      return provides, nil
+    }
+    ```
+
+We want to test that our component updates the `started` and `stopped` fields. 
+
+To accomplish it we are going to create a new lifecycle instance, create a `Require` struct instance, intialize our component, and validate that calling `Start` on the lifecycle instace to validate that the component hook has being called and the logic has being execuited.
+
+To create a lifecycle instance, we will use a helper function. `compdef.NewTestLifecycle()`. The function returns a lyfecycle wrapper that we can use to populate our `Requires` struct. Also, we can  call the `Start` and `Stop` functions.
+
+!!! Info 
+    <!-- TODO add link to NewTestLifecycle function once this PR https://github.com/DataDog/datadog-agent/pull/25184 is merged-->
+    You can see the `NewTestLifecycle` function [here]()
+
+=== ":octicons-file-code-16: comp/someomponent/impl/component.go"
+    ```go
+    package impl
+
+    import (
+      "context"
+      "testing"
+
+      compdef "github.com/DataDog/datadog-agent/comp/def"
+      "github.com/stretchr/testify/assert"
+    )
+
+    func TestStartHook(t *testing.T) {
+      lc := compdef.NewTestLifecycle()
+      
+      requires := Requires{
+        Lc:  lc,
+      }
+
+      provides, err := NewComponent(requires)
+
+      assert.NoError(t, err)
+
+      assert.NotNil(t, provides.Comp)
+      internalComponent := provides.Comp.(*component)
+
+      ctx := context.Background()
+      assert.NoError(t, lc.Start(ctx))
+
+      assert.True(t, internalComponent.started)
+    }
+    ```
+    
+For our example we had to perform a type cast operation, because the `started` field is private. Dependeding our your component you might not have to do that.
```

Comment:
```suggestion
For this example, a type cast operation had to be performed because the `started` field is private. Depending on the component, this may not be necessary.
```